src/collection.js

```diff
@@ -1422,6 +1439,19 @@ export class CollectionTransaction {
     return { data: existing, permissions: {} };
   }
 
+  /**
+   * Soft Delete all records from the local database.
+   * @param  {Object} allRecords       All the records in the collection.
+   * @return {Object}
+   */
+  deleteAll(allRecords) {
+    allRecords.forEach(record => {
+      delete record.id;
```

Comment:
this won't work, it only deletes the `id` property from the `record` object

Reply:
if I recall the delete method takes the record Id, looks it up and set the `_status` to delete (default option is soft delete) so theoretically, it should work and the unit test agrees!

Reply:
But here you don't call the `delete()` method, you use the `delete` keyword which does something else :)

Reply:
I'm calling the delete method, it was formatted this way by `cs-format` for some reason 😕